src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState_Checksum.cs

```diff
@@ -36,11 +36,9 @@ private async Task<DocumentStateChecksums> ComputeChecksumsAsync(CancellationTok
         {
             using (Logger.LogBlock(FunctionId.DocumentState_ComputeChecksumsAsync, FilePath, cancellationToken))
             {
-                var serializer = solutionServices.GetRequiredService<ISerializerService>();
-
                 var infoChecksum = this.Attributes.Checksum;
                 var serializableText = await SerializableSourceText.FromTextDocumentStateAsync(this, cancellationToken).ConfigureAwait(false);
-                var textChecksum = serializer.CreateChecksum(serializableText, cancellationToken);
+                var textChecksum = serializableText.ContentChecksum;
```

Comment:
this allows us to funnel along the checksum in the case where the serializable text is still what we're pointing at.  Note, in the case it got dumped as well to a memory mapped file on the OOP side, we'll dump the checksum at that point too, which should mean we're still efficient without ahving to realize the text at all.

Note: i'd actually like to take things further here.  Specifically, have us *not* dump SerializableSourceTexts to memory mapped files on the OOP side.  Since they already link to segments of a memory mapped file in the host, we can just keep them as is